kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -663,7 +663,7 @@ ImmutableOutput KeyValueBlockchain::handleCategoryUpdates(BlockId block_id,
 }
 
 /////////////////////// state transfer blockchain ///////////////////////
-void KeyValueBlockchain::addRawBlock(const RawBlock& block, const BlockId& block_id) {
+void KeyValueBlockchain::addRawBlock(const RawBlock& block, const BlockId& block_id, bool trylinkSTChainFrom) {
```

Comment:
why `try` and not just `link`? and how is it used?

Reply:
Link is the default case, and not optimized.
Let's say we put 50,000 blocks.
code always try to link. This is costly: it calls linkSTChainFrom, which calls getRawBlock and more operations. I can skip all that easily by calling 49,999 times with false (do not try to link), and only on the last time with true (link). when putting the last block. Its a very easy change and it saves some unneeded operations.